packages/klighd-core/src/skgraph-models.ts

```diff
@@ -36,13 +40,58 @@ export const LABEL_TYPE = 'label'
 /**
  * Represents the Sprotty version of its java counterpart in KLighD.
  */
-export class SKNode extends KNode {
+export class SKNode extends KNode implements SKGraphElement {
     tooltip?: string
+
+    private _node_scaled_bounds?: {bounds: Bounds, scale: number, effective_child_zoom: number}
+
     hasFeature(feature: symbol): boolean {
         return feature === selectFeature
             || (feature === moveFeature && (this.parent as SKNode).properties && (this.parent as SKNode).properties.interactiveLayout)
             || feature === popupFeature
     }
+
+    forceNodeScaleBounds(ctx: SKGraphModelRenderer, force = false): {bounds: Bounds, scale: number, effective_child_zoom: number}{
```

Comment:
I have now added some comments.
Based on <https://github.com/eclipse/sprotty/wiki/Architectural-Overview> features are for user interactions with diagrams. I don't think this classifies a a user interaction, as the user interacts with the viewport zoom and pan rather than this directly.